src/android/toga_android/widgets/box.py

```diff
@@ -1,8 +1,18 @@
 from .base import Widget
 from ..libs.activity import MainActivity
-from ..libs.android_widgets import RelativeLayout
+from ..libs.android_widgets import RelativeLayout, RelativeLayout__LayoutParams
 
 
 class Box(Widget):
     def create(self):
         self.native = RelativeLayout(MainActivity.singletonThis)
+
+    def set_child_bounds(self, widget, x, y, width, height):
+        # We assume `widget.native` has already been added to this `RelativeLayout`.
+        #
+        # We use `topMargin` and `leftMargin` to perform absolute layout. Not very
+        # relative, but that's how we do it.
+        layout_params = RelativeLayout__LayoutParams(width, height)
+        layout_params.topMargin = y
+        layout_params.leftMargin = x
+        self.native.updateViewLayout(widget.native, layout_params)
```

Comment:
I'm guessing this is a case where we can't re-use the existing object and update; we have to replace every time?

Reply:
Yeah -- from what I can tell, `LayoutParams` in general are intended to be treated as immutable, disposable Java objects. This one has public `topMargin` and `leftMargin` properties, but as far as I can tell, that's just to help you create it; mutating those didn't seem to have any effect from what I recall in my Kotlin testing. I can re-validate that if you like.

Reply:
If it's not too much trouble, it's worth re-validating; but it also makes sense in the Java context that it would be 'Use and discard".

Reply:
Done.

Reply:
I just tested from Kotlin, and if I mutate the props, it doesn't cause any change in the layout.

<details>
<summary>
Sample code -- no change in layout in 2 seconds
</summary>

```kotlin
package org.asheesh.fourbuttonsattwohundredpixels

import android.os.Bundle
import android.os.Handler
import android.util.DisplayMetrics
import android.view.View
import android.widget.Button
import android.widget.RelativeLayout
import android.widget.ScrollView
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    private lateinit var button1: Button
    private lateinit var button2: Button
    private lateinit var scrollThing: ScrollView
    private lateinit var dynamicLayout: RelativeLayout
    private lateinit var button_2_props: RelativeLayout.LayoutParams
    private var handler: Handler = Handler()


    override fun onCreate(savedInstanceState: Bundle?) {
        dynamicLayout = RelativeLayout(this)
        scrollThing = ScrollView(this)
        scrollThing.addView(dynamicLayout)
        setContentView(scrollThing)
        val displayMetrics = DisplayMetrics()
        windowManager.defaultDisplay.getMetrics(displayMetrics)
        val height: Int = displayMetrics.heightPixels
        val width: Int = displayMetrics.widthPixels
        println("width is ${width}")
        println("height is ${height}")

        button1 = Button(this)
        button1.text = "Kale chips food truck pop-up distillery prism. Craft beer art party copper mug shaman whatever quinoa try-hard synth meditation vexillologist mixtape readymade. Poutine microdosing keffiyeh, offal 8-bit chia twee. Salvia flexitarian coloring book sriracha meggings microdosing brunch vaporware craft beer. Fam green juice everyday carry, pitchfork forage retro health goth. Banjo messenger bag mlkshk VHS mumblecore austin single-origin coffee la croix. Whatever umami lumbersexual poutine organic marfa mustache raclette yuccie try-hard kickstarter tumblr cliche brooklyn food truck."
        val atMostFourHundred = View.MeasureSpec.makeMeasureSpec(400, View.MeasureSpec.AT_MOST);
        button1.measure(atMostFourHundred, View.MeasureSpec.UNSPECIFIED)
        println("Button1 measuredWidth after measure() ${button1.measuredWidth}")

        button2 = Button(this)
        button2.text = "Zounds"
        button2.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED)
        println("Button2 measuredWidth after measure() ${button2.measuredWidth}")

        //println("Button1 width ${button1.width}")
        dynamicLayout.addView(button1, RelativeLayout.LayoutParams(button1.measuredWidth, button1.measuredHeight))
        val params = RelativeLayout.LayoutParams(button2.measuredWidth, button2.measuredHeight)
        button_2_props = params
        params.topMargin = 2000
        params.leftMargin = button1.measuredWidth + 70
        dynamicLayout.addView(button2, params )

        //println("Button1 width after addView ${button1.width}")
        //println("Button1 measuredWidth after addView ${button1.measuredWidth}")
        //println("Button1 width after measure() ${button1.width}")
        //println("Button1 maxWidth after measure() ${button1.maxWidth}")
        super.onCreate(savedInstanceState)
    }

    override fun onResume() {
        super.onResume()
        println("Button1 width in onResume ${button1.width}")
        println("Button1 measuredWidth in onResume ${button1.measuredWidth}")
        button1.measure(View.MeasureSpec.AT_MOST, View.MeasureSpec.AT_MOST)

        // Move button2 higher up, 2 sec later
        handler.postDelayed(Runnable {
            button_2_props.topMargin = 500
            button_2_props.leftMargin = button1.measuredWidth + 70
            val params = RelativeLayout.LayoutParams(button2.measuredWidth, button2.measuredHeight)
        }, 2000)


    }
}
```

</details>